### PR TITLE
Bug 1881816: Gather MachineSet info

### PR DESCRIPTION
**Pull request by tremes**

### CODE REVIEW

**Comment by tremes** [comment]
Can you please review @natiiix, @martinkunc ?

**Comment by 0sewa0** [comment]
lgtm

File: docs/gathered-data.md [hunk 1]
```diff
@@ -115,13 +128,6 @@ See: docs/insights-archive-sample/config/clusteroperator
 Location of pods in archive: config/pod/
 
 
-Output raw size: 245
-
-### Examples
-
-#### ClusterOperators
-[{"Name":"config/clusteroperator/","Captured":"0001-01-01T00:00:00Z","Fingerprint":"","Item":{"metadata":{"creationTimestamp":null},"spec":{},"status":{"conditions":[{"type":"Degraded","status":"","lastTransitionTime":null}],"extension":null}}}]
-
```
Comment (tremes) [inline docs/gathered-data.md:124]: Not sure why this is removed

File: pkg/gather/clusterconfig/clusterconfig_test.go [hunk 2]
```diff
@@ -537,6 +541,42 @@ func TestGatherHostSubnet(t *testing.T) {
...
+	if len(errs) > 0 {
+		t.Errorf("unexpected errors: %#v", errs)
+		return
+	}
+	if len(records) != 1 {
+		t.Fatalf("unexpected number or records %d", len(records))
+	}
+	if records[0].Name != "machinesets/test-worker" {
+		t.Fatalf("unexpected machineset name %s", records[0].Name)
+	}
```
Comment (tremes) [inline pkg/gather/clusterconfig/clusterconfig_test.go:577]: This is pretty simple test, but I was struggling with unmarshalling the record to something meaningful (wasn't successful with `Unstructured`).

File: docs/insights-archive-sample/machinesets/d50d0126-c90b-4428-a75f-dc08cd02960a-worker-test [hunk 3]
```diff
@@ -0,0 +1 @@
...chine.openshift.io/cluster-api-cluster":"d50d0126-c90b-4428-a75f-dc08cd02960a","machine.openshift.io/cluster-api-machine-role":"worker","machine.openshift.io/cluster-api-machine-type":"worker","machine.openshift.io/cluster-api-machineset":"d50d0126-c90b-4428-a75f-dc08cd02960a-worker-test"}},"spec":{"metadata":{"labels":{"node-role.kubernetes.io/\u003crole\u003e":""}},"providerSpec":{"value":{"ami":{"id":"ami-046fe691f52a953f9"},"apiVersion":"awsproviderconfig.openshift.io/v1beta1","blockDevices":[{"ebs":{"iops":0,"volumeSize":120,"volumeType":"gp2"}}],"credentialsSecret":{"name":"aws-cloud-credentials"},"deviceIndex":0,"iamInstanceProfile":{"id":"d50d0126-c90b-4428-a75f-dc08cd02960a-worker-profile"},"instanceType":"m4.large","kind":"AWSMachineProviderConfig","placement":{"availabilityZone":"us-east-1a","region":"us-east-1"},"securityGroups":[{"filters":[{"name":"tag:Name","values":["d50d0126-c90b-4428-a75f-dc08cd02960a-worker-sg"]}]}],"subnet":{"filters":[{"name":"tag:Name","values":["d50d0126-c90b-4428-a75f-dc08cd02960a-worker-test"]}]},"tags":[{"name":"kubernetes.io/cluster/d50d0126-c90b-4428-a75f-dc08cd02960a","value":"owned"}],"userDataSecret":{"name":"worker-user-data"}}}}}}}}
```
Comment (tremes) [inline docs/insights-archive-sample/machinesets/d50d0126-c90b-4428-a75f-dc08cd02960a-worker-test:1]: I think I need to fix this. There shouldn't be any `Object`

**Comment by martinkunc** [comment]
Great PR, thanks Tomas,

/lgtm

**Pull request merged**